Library/Homebrew/cmd/install.rb

```diff
@@ -338,21 +336,67 @@ def run
           skip_link:                  args.skip_link?,
         )
 
-        Upgrade.check_installed_dependents(
-          installed_formulae,
-          flags:                      args.flags_only,
-          installed_on_request:       !args.as_dependency?,
-          force_bottle:               args.force_bottle?,
-          build_from_source_formulae: args.build_from_source_formulae,
-          interactive:                args.interactive?,
-          keep_tmp:                   args.keep_tmp?,
-          debug_symbols:              args.debug_symbols?,
-          force:                      args.force?,
-          debug:                      args.debug?,
-          quiet:                      args.quiet?,
-          verbose:                    args.verbose?,
-          dry_run:                    args.dry_run?,
-        )
+        if args.ask?
+          dependants = Upgrade.get_dependants(
+            installed_formulae,
+            flags:                      args.flags_only,
+            ask:                        args.ask?,
+            installed_on_request:       !args.as_dependency?,
+            force_bottle:               args.force_bottle?,
+            build_from_source_formulae: args.build_from_source_formulae,
+            interactive:                args.interactive?,
+            keep_tmp:                   args.keep_tmp?,
+            debug_symbols:              args.debug_symbols?,
+            force:                      args.force?,
+            debug:                      args.debug?,
+            quiet:                      args.quiet?,
+            verbose:                    args.verbose?,
+            dry_run:                    args.dry_run?,
+          )
+
+          formulae_dependencies = Install.get_hierarchy(formulae_installer, dependants)
+          # Main block: if asking the user is enabled, show dependency and size information.
+          Install.ask_formulae(formulae_dependencies, args: args)
+        end
+
+        Install.install_formulae(formulae_installer,
+                                 dry_run: args.dry_run?,
+                                 verbose: args.verbose?)
+
+        unless args.ask?
+          dependants = Upgrade.get_dependants(
```

Comment:
```suggestion
          upgrade_dependants = Upgrade.get_dependants(
```
maybe? Makes it clearer why we're calling `Upgrade.*` here